### PR TITLE
[MMCA-4190] - CDS Financials to create a new endpoint to receive notifications from CDDM

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Code changes for new Endpoint

Below has been done as part of this PR

- New controller and tests
- Request, response schemas and classes for new request and responses
- New class/trait for headers
- New service and cache method for update operation
- Use of Enum for search status

### CODE REVIEW

File: test/resources/ssfn-invalid-error-response.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "errorCode": "400",
+    "errorMessage": "Bad request received",
+    "source": "CDS Financials",
+    "sourceFaultDetail": {
+      "detail": [
+        "Invalid value supplied for field statementRequestId: 32"
+      ]
+    }
+  }
+}
```
Comment (AdamCoops) [inline test/resources/ssfn-invalid-error-response.json:14]: We're introducing a few end of files errors in the .json files, does this matter?

Comment (tyagikumarvineet) [reply test/resources/ssfn-invalid-error-response.json]: Done.

File: test/models/requests/test.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "errorDetail": { "timestamp": "2019-08-16T18:15:41Z", "correlationId": "3jh1f6b3-f8b1-4f3c-973a-05b4720e", "errorCode": "400", "errorMessage": "Bad request received", "source": "Backend", "sourceFaultDetail": { "detail": [ "Invalid value supplied for field statementRequestId: 32 | Invalid value supplied for field failureReason: NoDocs " ] } }}
```
Comment (AdamCoops) [inline test/models/requests/test.json:1]: Here

File: conf/schemas/statement-search-failure-notification-request-schema.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        "reason"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "required": [
+    "StatementSearchFailureNotificationMetadata"
+  ],
+  "additionalProperties": false
+}
```
Comment (AdamCoops) [inline conf/schemas/statement-search-failure-notification-request-schema.json:37]: Here

File: conf/schemas/statement-search-failure-notification-error-response-schema.json [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        "source"
+      ],
+      "additionalProperties": false
+    }
+  },
+  "required": [
+    "errorDetail"
+  ],
+  "additionalProperties": false
+}
```
Comment (AdamCoops) [inline conf/schemas/statement-search-failure-notification-error-response-schema.json:58]: Here

File: test/models/requests/test.json (hunk 2 above)
Comment (tyagikumarvineet) [reply test/models/requests/test.json]: Done, file is not needed. hence removed.

File: conf/schemas/statement-search-failure-notification-request-schema.json (hunk 3 above)
Comment (tyagikumarvineet) [reply conf/schemas/statement-search-failure-notification-request-schema.json]: Done

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**